llvm/test/CodeGen/AArch64/aarch64-neon-vector-insert-uaddlv.ll

```diff
@@ -146,12 +146,13 @@ define void @insert_vec_v6i64_uaddlv_from_v4i32(ptr %0) {
 ; CHECK-LABEL: insert_vec_v6i64_uaddlv_from_v4i32:
 ; CHECK:       ; %bb.0: ; %entry
 ; CHECK-NEXT:    movi.2d v0, #0000000000000000
-; CHECK-NEXT:    str xzr, [x0, #16]
```

Comment:
Can we bail from the combine when storing zero?

Reply:
I don't think that will prevent this, this first changed to `str xzr` with my previous patch #129756, so it think it's more something not recognizing the sub register extract.

Reply:
I've updated the combine to propagate zeros in cases like this (with restores this test) -- just bailing out is not enough.